front/app/components/PostShowComponents/Comments/CommentSorting.tsx

```diff
@@ -1,55 +1,61 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import FilterSelector from 'components/FilterSelector';
-import { FormattedMessage } from 'utils/cl-intl';
+import { FormattedMessage, MessageDescriptor } from 'utils/cl-intl';
 import messages from './messages';
-import styled from 'styled-components';
 import { CommentsSort } from 'api/comments/types';
-
-const Container = styled.div``;
+import { Box } from '@citizenlab/cl2-component-library';
 
 interface Props {
-  id?: string | undefined;
   onChange: (value: CommentsSort) => void;
-  selectedValue: CommentsSort[];
+  selectedCommentSort: CommentsSort;
   className?: string;
 }
 
-interface State {}
-
-export default class CommentSorting extends PureComponent<Props, State> {
-  handleOnChange = (selectedValue: [CommentsSort]) => {
-    this.setState({ selectedValue });
-    this.props.onChange(selectedValue[0]);
+const CommentSorting = ({
+  onChange,
+  selectedCommentSort,
+  className,
+}: Props) => {
+  const handleOnChange = (selectedValue: ['new'] | ['-upvotes_count']) => {
```

Comment:
Minor: we can reuse `CommentsSort` here.

```suggestion
  const handleOnChange = (selectedValue: [CommentsSort]) => {
```

Reply:
👍 Actually not minor. I was missing a type after the last changes.